### PR TITLE
build: Refactor VkLayer_util install files

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
- Refactor install destination to match source
- Remove layers/generated from include directories

### CODE REVIEW

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -73,7 +73,40 @@ endif()
 if(BUILD_LAYER_SUPPORT_FILES)
     # NOTE: We need to install header/source files for Android
     get_target_property(LAYER_UTIL_FILES VkLayer_utils SOURCES)
-    install(FILES ${LAYER_UTIL_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/vulkan)
```
Comment (ncesario-lunarg) [inline layers/CMakeLists.txt:76]: This apparently flattens the directory structure, so we can't use this. If there's a way to preserve the directory structure, we could possibly do that, but then we would need to do something custom for the `generated` files. In general, being explicit with install files seems like the safest thing to do.

**Review by juan-lunarg — COMMENTED** [review]
This doesn't take into account fixing the Android builds. See the code in the layer factory. They also reference the directory structure of our project. Not just our install directory.

File: layers/CMakeLists.txt [hunk 2]
```diff
@@ -67,13 +64,42 @@ if (USE_ROBIN_HOOD_HASHING)
     target_compile_definitions(robin_hood::robin_hood INTERFACE USE_ROBIN_HOOD_HASHING)
 endif()
 
-# Currently repos like LunarG/VulkanTools require source code from VVL
-# This results in VkLayer_utils needing to be installed. Ideally this is
-# cleaned up in the future since this is a result of history involving the mono-repo.
+# NOTE: We need to install header/source files for Android
+# TODO: This should be removed once the official "Vulkan Layer Utils" library is available for consumption
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:68]:
```suggestion
# TODO: This should be removed once the official "Vulkan-Utility-Libraries" library is available for consumption
```

File: .gitignore [hunk 3]
```diff
@@ -39,6 +39,7 @@ build-android/third_party
 .editorconfig
 scripts/.vs
 .*.sw[op]
+*~
```
Comment (juan-lunarg) [inline .gitignore:42]: ?

```suggestion
```

**Review by juan-lunarg — APPROVED** [review]
LGTM

**Comment by juan-lunarg** [comment]
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/Vulkan-ValidationLayers?pullRequest=5494) before we can accept your contribution.

It looks like someone using the GitHub username nces has committed to this PR and would need to sign also.

File: .gitignore (hunk 3 above)
Comment (ncesario-lunarg) [reply .gitignore:42]: Vim temporary files on some old versions of vim.

**Pull request merged**